File: templates/Uwp/Features/WhatsNewPrompt.Prism/Services/WhatsNewDisplayService.cs
using System;
using System.Threading.Tasks;
using Microsoft.Toolkit.Uwp.Helpers;
using Windows.ApplicationModel.Core;
using Windows.UI.Core;
using Param_RootNamespace.Views;

namespace Param_RootNamespace.Services
{
    // For instructions on testing this service see https://github.com/Microsoft/WindowsTemplateStudio/tree/master/docs/features/whats-new-prompt.md
    public class WhatsNewDisplayService : IWhatsNewDisplayService
    {
        private static bool shown = false;

        public async Task ShowIfAppropriateAsync()
        {
            await CoreApplication.MainView.CoreWindow.Dispatcher.RunAsync(
                CoreDispatcherPriority.Normal, async () =>
                {
                    if (SystemInformation.IsAppUpdated && !shown)
                    {
                        shown = true;
                        var dialog = new WhatsNewDialog();
                        await dialog.ShowAsync();
                    }
                });
        }
    }
}

File: templates/Uwp/Features/WhatsNewPromptVB/Services/WhatsNewDisplayService.vb
Imports Param_RootNamespace.Views
Imports Microsoft.Toolkit.Uwp.Helpers

Namespace Services
    ' For instructions on testing this service see https://github.com/Microsoft/WindowsTemplateStudio/tree/master/docs/features/whats-new-prompt.md
    Public Module WhatsNewDisplayService
        Dim shown As Boolean = False

        Friend Async Function ShowIfAppropriateAsync() As Task
            If SystemInformation.IsAppUpdated AndAlso Not shown Then
                shown = True
                Dim dialog = New WhatsNewDialog()
                Await dialog.ShowAsync()
            End If
        End Function
    End Module
End Namespace
